### PR TITLE
Add dynamic layout spacing

### PR DESCRIPTION
**Pull request by unscripted**
## Changes proposed in this pull request

This PR implements a robust layout system in `theme.json` that handles responsive design and content widths. 

Key features include:

### Custom Layout Properties
- `full-viewport-width`: Calculates viewport width minus scrollbar (set to 0px to prevent layout shifts)
- `site-content-width`: Determines main content area width with responsive padding
- `site-outer-padding`: Uses `max()` to ensure proper spacing on devices with notches/rounded corners
- Side spacing calculations for both regular and wide content blocks

### Layout Settings
- `contentSize`: Caps regular content at 680px while maintaining responsiveness
- `wideSize`: Allows wider content blocks up to 1200px

The system uses CSS custom properties for flexibility and maintainability, ensuring consistent spacing and responsive behavior across different screen sizes and devices.

### 1. Layout Testing
- [ ] Test on different screen sizes (mobile, tablet, desktop)
- [ ] Verify content width is capped at 680px for regular content
- [ ] Verify wide content blocks expand up to 1200px
- [ ] Check that content remains centered with equal margins
- [ ] Test on devices with notches/rounded corners to verify safe-area spacing

### 2. Responsive Behavior
- [ ] Test the following breakpoints:
  - Mobile: 320px - 767px
  - Tablet: 768px - 1023px
  - Desktop: 1024px+
- [ ] Verify no horizontal scrollbars appear
- [ ] Check that spacing scales appropriately at each breakpoint
- [ ] Test with different content lengths (short and long)

### 3. Browser Compatibility
- [ ] Test in:
  - Chrome (latest)
  - Firefox (latest)
  - Safari (latest)
  - Edge (latest)
- [ ] Verify layout calculations work consistently across browsers
- [ ] Check that scrollbar behavior is consistent

### 4. Content Types
- [ ] Test with:
  - Regular paragraphs
  - Images (various sizes)
  - Block editor elements
  - Custom blocks
  - Full-width elements
  - Wide-width elements

### 5. Edge Cases
- [ ] Test with:
  - Very long words/URLs
  - Tables
  - Code blocks
  - Embedded content
  - Custom HTML
  - RTL languages

### 6. Accessibility
- [ ] Verify content remains readable at all sizes
- [ ] Check that spacing doesn't interfere with readability
- [ ] Test with screen readers
- [ ] Verify keyboard navigation works properly

Once testing is complete, notify the author of any failed tests and move the task to "Kick back" in Asana. If all tests pass, move the task to "Ready for Code Review" in Asana and tag a team member for code review.

### Code review

As the code reviewer for this pull request, I verify that:

- [ ] All automated tests have passed.
- [ ] The code is written (or documented) in a way that is easy to understand.
- [ ] The code is free of obvious errors and duplication.
- [ ] The code follows our coding standards.
- [ ] The code is sanitized or escaped appropriately for any SQL or XSS injection possibilities.

Once testing is complete, notify the author of any failed tests and move the task to "Kick back" in Asana or continue with the "merging" steps below.

## Merging

As the individual merging this pull request, I verify that:

- [ ] All automated tests have passed.
- [ ] All functional tests have passed.
- [ ] All code review tests have passed.
- [ ] I have moved the task to "Ready to Deploy" in Asana and notified the pull request author.

### CODE REVIEW

File: theme.json [hunk 1]
```diff
@@ -2,9 +2,19 @@
...
+				"scrollbar-width": "0px",
+				"site-content-width": "calc(var(--wp--custom--layout--full-viewport-width) - (2 * var(--wp--custom--layout--site-outer-padding)))",
+				"site-outer-padding": "max(var(--wp--preset--spacing--16, 1rem), env(safe-area-inset-left))"
+			}
+		},
 		"layout": {
-			"contentSize": "1200px",
-			"wideSize": "1440px"
+			"contentSize": "min(680px, var(--wp--custom--layout--site-content-width))",
+			"wideSize": "min(1200px, var(--wp--custom--layout--site-content-width))"
```
Comment (sstruemph) [inline theme.json:17]: @unscripted Why?

Every project we've build at least for the last few years has been 1200px content width and 1440px wide width.

File: theme.json [hunk 2]
```diff
@@ -2,9 +2,19 @@
...
 	"settings": {
 		"appearanceTools": true,
+		"custom": {
+			"layout": {
+				"full-viewport-width": "calc(100vw - var(--wp--custom--layout--scrollbar-width, 0px))",
+				"main-content-width-side-spacing": "calc((var(--wp--custom--layout--full-viewport-width) - var(--wp--style--global--content-size)) / 2)",
+				"main-wide-width-side-spacing": "calc((var(--wp--custom--layout--full-viewport-width) - var(--wp--style--global--wide-size)) / 2)",
+				"scrollbar-width": "0px",
+				"site-content-width": "calc(var(--wp--custom--layout--full-viewport-width) - (2 * var(--wp--custom--layout--site-outer-padding)))",
+				"site-outer-padding": "max(var(--wp--preset--spacing--16, 1rem), env(safe-area-inset-left))"
```
Comment (sstruemph) [inline theme.json:12]: @unscripted  This might introduce issues elsewhere but I can't really tell. What was the reasoning for trying out these styles? Has this been tested on a production site?
